### ISSUE TITLE
Avoid pulling in multiple versions of the syn crate

### ISSUE THREAD

**Issue by Bathtor** [JoshMcguigan/arr_macro#5]
Hi,

I was doing some cleaning up of dependencies on one of my crates, and I noticed that your crate (which is a transitive dependency for me) pulls in two different versions on the `syn` crate: `0.15.44` directly, and `1.0.7` via `proc-macro-hack`.

So, I would like to make a feature request to update this crate to use syn `1.0` so that it doesn't pull in multiple versions.

It's not a huge issue, of course, no rush or anything. Just a nice little cleanup thing :)

Cheers,
Lars

**Comment by Bathtor** [comment]
My resolution tree for reference:

```
syn v0.15.44
└── arr_macro_impl v0.1.2
    └── arr_macro v0.1.2
        └── executors v0.5.3
            └── kompact v0.8.1 (/Users/lkroll/Documents/Programming/Kompics/kompics-rust/core)

syn v1.0.7
├── kompact-actor-derive v0.8.1 (/Users/lkroll/Documents/Programming/Kompics/kompics-rust/macros/actor-derive)
│   └── kompact v0.8.1 (/Users/lkroll/Documents/Programming/Kompics/kompics-rust/core)
├── kompact-component-derive v0.8.1 (/Users/lkroll/Documents/Programming/Kompics/kompics-rust/macros/component-definition-derive)
│   └── kompact v0.8.1 (/Users/lkroll/Documents/Programming/Kompics/kompics-rust/core) (*)
└── proc-macro-hack v0.5.11
    ├── arr_macro v0.1.2
    │   └── executors v0.5.3
    │       └── kompact v0.8.1 (/Users/lkroll/Documents/Programming/Kompics/kompics-rust/core) (*)
    └── arr_macro_impl v0.1.2
        └── arr_macro v0.1.2 (*)
```

**Comment by JoshMcguigan** [comment]
Thanks for the suggestion. I've just published version `0.1.3` of this crate, which relies on version `1.0` of syn.
